src/spi.rs

```diff
@@ -0,0 +1,166 @@
+use {Register, Pin};
+
+
+/// An SPI module.
+///
+/// Information at
+/// http://maxembedded.com/2013/11/the-spi-of-the-avr/
+pub trait HardwareSpi {
+    /// Master-in slave-out pin.
+    type MISO: Pin;
```

Comment:
I don't know if this is a losing battle, but I really was hoping we could avoid having acronym soup. I'm happy that there are comments, but I would rather it be 

```rust
type MasterInSlaveOut: Pin;
```

And now the comment is encoded in the code. We can then add comments describing the why and the how, instead of the *what*.

The reason I think it might be a losing battle is because there's going to be inertia from all sides to "just do what Arduino / C / Atmel does". The reason I don't think that's fully valid is because we want to do things as Rust-like as we can.

Reply:
I think this is a good idea.